File: C++ Code/Data Structure/Stack/Implementation using Linked List.cpp
// WAP to implement a stack using Linked List.
#include<iostream>
using namespace std;
class node{
  public:
    int data;
    node* next;
    
  node(int data){
    this->data=data;
    this->next=NULL;
  }  
};
class stack{
  public:
    node* head;
    int size;
    
  stack(){
    head=NULL;
    size=0;
  }
  void push(int data){
    node* temp=new node(data);
    if(temp){
      temp->next=head;
      head=temp;
      size++;
    }
    else
      cout<<endl<<"Stack Overflow."<<endl;
  }
  void pop(){
    if(size==0){
      cout<<endl<<"Stack Underflow"<<endl;
      return ;
    }
    node* temp=head;
    head=head->next;
    temp->next=NULL;
    delete temp;
    size--;
  }
  int length(){
    return size;
  }
  bool isEmpty(){
    return size==0;
  }
  int top(){
    return head->data;
  }
};
int main(){
    stack st;
    int x;
    cout<<"Enter 1 for push operation or 2 for pop operation : ";
    cin>>x;
    while(x==1 ||x==2){
      if(x==1){
        int data;
        cout<<"Enter data : ";
        cin>>data;
        st.push(data);
      }
      else if(x==2)
        st.pop();
      else
        break;
      if(st.isEmpty())
        cout<<endl<<"Stack is empty."<<endl;
      else{
        cout<<"Top element : "<<st.top()<<endl;
        cout<<"Size of Stack : "<<st.length()<<endl;
      }
      cout<<"Enter 1 for push operation or 2 for pop operation, other to stop opertaion : ";
      cin>>x;
    }
  return 0;
}

File: Javascript Code/Stack/Implementation using Linked List.js
class Node{
  constructor(data){
    this.data=data===undefined?0:data;
    this.next=null;
  }
}
class Stack{
  constructor(){
    this.head=null;
    this.size=0;
  }
  push(data){
    let temp=new Node(data);
    temp.next=this.head;
    this.head=temp;
    this.size++;
  }
  pop(){
    if(this.size===0)
      throw new Error("Stack underflow");
    this.head=this.head.next;
    this.size--;
  }
  peek(){
    if(this.size===0)
      throw new Error("Stack underflow");
    return this.head.data;
  }
  isEmpty(){
    return this.size===0;
  }
  Size(){
    return this.size;
  }
  isPresent(val){
    for(let i=this.head;i!==null;i=i.next){
      if(i.data===val)
        return true;
    }
    return false;
  }
}